controllers/datadogagent/controller_reconcile_v2_common.go

```diff
@@ -154,7 +154,13 @@ func (r *Reconciler) createOrUpdateDaemonset(parentLogger logr.Logger, dda *data
 		// check if same hash
 		needUpdate := !comparison.IsSameSpecMD5Hash(hash, currentDaemonset.GetAnnotations())
 		if !needUpdate {
-			// no need to update to stop here the process
+			// Even if the DaemonSet is still the same, it's status might have
+			// changed (for example, the number of pods ready). This call is
+			// needed to keep the agent status updated.
+			now := metav1.NewTime(time.Now())
```

Comment:
should we pass `now` as input of the `createOrUpdateDaemonset` function?

Reply:
I'm not sure. The function already has many params and `now` is easily accessible from here.
If it made testing easier, for example, we could do it. But I don't see a big benefit now. No strong opinion.